src/TributeSearch.js

```diff
@@ -68,10 +68,15 @@ class TributeSearch {
         let best, temp
 
         while (index > -1) {
-            patternCache.push(index)
-            temp = this.traverse(string, pattern, index + 1, patternIndex + 1, patternCache)
-            patternCache.pop()
-
+            if (!this.tribute.exactMatch || patternCache.length < 1) {
+              patternCache.push(index)
+              temp = this.traverse(string, pattern, index + 1, patternIndex + 1, patternCache)
+              patternCache.pop()
+            } else if (this.tribute.exactMatch && patternCache[patternCache.length -1] + 1 === index) {
+              patternCache.push(index)
+              temp = this.traverse(string, pattern, index + 1, patternIndex + 1, patternCache)
+              patternCache.pop()
+            }
```

Comment:
I would refactor this so we can avoid duplicating the business logic here. One idea could be to create a new method does the checks and have this method only performing the operation:

```javascript

function shouldWeDoWork(patternCache, index) {
    if (!this.trubute.exactMatch || patternCache.length < 1) {
         return true;
    }
    if (this.tribute.exactMatch && patternCache[patternCache.length - 1] + 1 === index) {
         return true;
    }
    return false;
}
// ...
while (index > -1) {
     if (this.shouldWeDoWork(patternCache, index) {
          patternCache.push(index)
          temp = this.traverse(string, pattern, index + 1, patternIndex + 1, patternCache)
          patternCache.pop()
     }
}
```

Reply:
(Obviously, you'd have to think of a proper name for that method and I think there's room for simplifying those if statements.)

Reply:
Yeah I was thinking about adding a separate function as well. Ill go ahead and add one I agree the duplication is 'code smell'

Reply:
Just pushed the update to this, moving the logic to a separate function